### PR TITLE
Offline mode

### PR DESCRIPTION
**Pull request by sslaws**
# Fixes or Implements [AB#9834](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9834)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description

Adds the ability to take web client offline.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES | NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

File: Testing/functional/e2e/cypress/integration/pages/LandingPage.js [hunk 1]
```diff
@@ -22,4 +22,21 @@ describe('Landing Page', () => {
                            .should('have.attr', 'href', '/login')
                            .should('have.text', 'Log in')
     })
+
```
Comment (leobcgov) [inline Testing/functional/e2e/cypress/integration/pages/LandingPage.js:25]: Add func. tests to the user story's PR

File: Apps/WebClient/src/ClientApp/src/router.ts [hunk 2]
```diff
@@ -62,15 +62,19 @@ enum UserState {
...
 
 function calculateUserState() {
+    const isOffline = store.getters["config/isOffline"];
     const isAuthenticated: boolean = store.getters["auth/oidcIsAuthenticated"];
     const isValid: boolean = store.getters["auth/isValidIdentityProvider"];
     const isRegistered: boolean = store.getters["user/userIsRegistered"];
     const userIsActive: boolean = store.getters["user/userIsActive"];
 
-    if (!isAuthenticated) {
+    if (isOffline) {
```
Comment (sslaws) [inline Apps/WebClient/src/ClientApp/src/router.ts:75]: Is the whitelist supported?

**Review by sslaws — COMMENTED** [review]
It doesn't seem like the whitelist is supported by the offline mode?

File: Apps/WebClient/src/ClientApp/src/router.ts (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/router.ts:75]: it is supported. The Vuex method checks it

File: Apps/WebClient/src/ClientApp/src/store/modules/config/getters.ts [hunk 3]
```diff
@@ -22,4 +23,26 @@ export const getters: GetterTree<ConfigState, RootState> = {
...
+            const startTime = new DateWrapper(offlineConfig.startDateTime);
+            const endTime = offlineConfig.endDateTime
+                ? new DateWrapper(offlineConfig.endDateTime)
+                : DateWrapper.fromNumerical(2050, 12, 31);
+
+            const now = new DateWrapper();
+            if (
+                now.isAfterOrSame(startTime) &&
+                now.isBeforeOrSame(endTime) &&
+                !offlineConfig.whitelist.includes(clientIP)
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/store/modules/config/getters.ts:41]: @sslaws here is the whitelist check

**Review by FuriousLlama — DISMISSED** [review]

File: Apps/WebClient/src/ClientApp/src/store/modules/config/getters.ts (hunk 3 above)
Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/store/modules/config/getters.ts:41]: right, this has passed both of my local and cypress' tests

**Review by leobcgov — DISMISSED** [review]

**Comment by leobcgov** [comment]
I'm fixing the conflicting issue

**Review by FuriousLlama — APPROVED** [review]

**Review by leobcgov — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**